protob/messages-tron.proto

```diff
@@ -0,0 +1,207 @@
+            optional bytes url = 3;                         // New URL
+        }
+        // Network proposal contract
+        message TronProposalCreateContract {
+            optional bytes owner_address = 1;               // Account address
+            map<uint64, uint64> parameters = 2;             // Parameter to be changed
+        }
+        // Approval contract
+        message TronProposalApproveContract {
+            optional bytes owner_address = 1;               // Account address
+            optional uint64 proposal_id = 2;                // Proposal ID
+            optional bool is_add_approval = 3;              // Add or remove approval
+        }
+        // Delete proposal
+        message TronProposalDeleteContract {
+            optional bytes owner_address = 1;               // Account address
+            optional uint64 proposal_id = 2;                // Proposal ID
+        }
+        // Update contract consume percentage
+        message TronUpdateSettingContract {
+            optional bytes owner_address = 1;               // Account address
+            optional bytes contract_address = 2;            // Contract Address
+            optional uint64 consume_user_resource_percent = 3;// Consume percentage
+        }
+        
+        optional TronTransferContract transfer_contract = 1;
+        optional TronTransferAssetContract transfer_asset_contract = 2;
+        optional TronVoteWitnessContract vote_witness_contract = 4;
+        optional TronWitnessCreateContract witness_create_contract = 5;
+        optional TronAssetIssueContract asset_issue_contract = 6;
+        optional TronWitnessUpdateContract witness_update_contract = 8;
+        optional TronParticipateAssetIssueContract participate_asset_issue_contract = 9;
+        optional TronAccountUpdateContract account_update_contract = 10;
+        optional TronFreezeBalanceContract freeze_balance_contract = 11;
+        optional TronUnfreezeBalanceContract unfreeze_balance_contract = 12;
+        optional TronWithdrawBalanceContract withdraw_balance_contract = 13;
+        optional TronUnfreezeAssetContract unfreeze_asset_contract = 14;
+        optional TronUpdateAssetContract update_asset_contract = 15;
+        optional TronProposalCreateContract proposal_create_contract = 16;
+        optional TronProposalApproveContract proposal_approve_ontract = 17;
+        optional TronProposalDeleteContract proposal_delete_contract = 18;
+        optional TronSetAccountIdContract set_account_id_contract = 19;   
+    }
+    // Common part of transaction
+    message TronTransaction {
+        optional bytes ref_block_bytes = 1;     // reference block number 
+        optional bytes ref_block_hash = 4;      // reference block hash
+        optional uint64 expiration = 8;         // transaction expiration
+        optional bytes data = 10;               // Extra transaction info
+        repeated TronContract contract = 11;    // contract messages
```

Comment:
if this is repeated, it should not be part of a single big `TronSignTx` message. Instead, make a `TronTxRequest` and `TronTxAck` that replies with the individual contract messages - see [this comment](https://github.com/trezor/trezor-common/pull/197#issuecomment-416635279) on EOS for the recommended flow.

Reply:
No need to be repeated, have changed to optional.